scripts/pi-hole/js/settings.js

```diff
@@ -334,15 +334,17 @@ $(".nav-tabs a").on("shown.bs.tab", function (e) {
 // Bar/Smooth chart toggle
 $(function () {
   var bargraphs = $("#bargraphs");
-  var chkboxData = localStorage.getItem("barchart_chkbox");
+  var chkboxData = localStorage ? localStorage.getItem("barchart_chkbox") : null;
 
   if (chkboxData !== null) {
     // Restore checkbox state
     bargraphs.prop("checked", chkboxData === "true");
   } else {
     // Initialize checkbox
     bargraphs.prop("checked", true);
-    localStorage.setItem("barchart_chkbox", true);
+    if (localStorage) {
+      localStorage.setItem("barchart_chkbox", true);
+    }
   }
 
   bargraphs.click(function () {
```

Comment:
note that I left the click functions for settings a storage object unguarded so there is a console error message to indicate the saving failed. I guarded initialization routines that set storage to prevent further code running during initialization from getting blocked.